### PR TITLE
Refactor/virtual register languages

### PR DESCRIPTION
**Pull request by martin-cs**
This makes a number of refactorings to the registration of languages.  They are in increasing order of controversy and I would suggest reviewing them commit by commit.

This connects to https://github.com/diffblue/cbmc/issues/6219 as it means that more tools are usable with the other front-ends.

### CODE REVIEW

**Comment by martin-cs** [comment]
@danielsn for RMC / `json_symtab` related changes.

**Comment by martin-cs** [comment]
One issue with this PR is that we will now have four files that are basically identical.  I had wanted to make this into the default implementation of `register_languages` but that makes `util` depend on the language front-ends which is not what we want.  Solutions to this that reduce this duplication would be most appreciated.

**Comment by kroening** [comment]
Please see my comment on #6219 -- there should not be further extension of CBMC to support non-C languages.

**Comment by martin-cs** [comment]
OK.  I have removed the last commit which does that.  Is the `symtab2gb` to `cbmc` route OK?

**Comment by martin-cs** [comment]
@tautschnig @peterschrammel @chrisr-diffblue this PR is now less controversial and is mostly a tidy up and refactor.  Any chance of a review when you have a moment?

**Comment by martin-cs** [comment]
@TGWDB As you are looking at https://github.com/diffblue/cbmc/issues/6219 it would be good to have your review on this.

**Review by TGWDB — APPROVED** [review]
This looks ok to me. I don't have significant comment on the question of how to reduce duplicated code in different places.

**Review by NlightNFotis — APPROVED** [review]
Solid refactoring 👍🏻

**Comment by martin-cs** [comment]
@chrisr-diffblue while you are reviewing things, any chance of a look at this?

File: src/memory-analyzer/memory_analyzer_parse_options.cpp
Comment (chrisr-diffblue) [inline src/memory-analyzer/memory_analyzer_parse_options.cpp:48]: Worth a reference to a GitHub issue here?

**Review by chrisr-diffblue — APPROVED** [review]

File: src/memory-analyzer/memory_analyzer_parse_options.cpp
Comment (martin-cs) [reply src/memory-analyzer/memory_analyzer_parse_options.cpp:48]: I didn't know there was one!  I'll have a look...

Comment (martin-cs) [reply src/memory-analyzer/memory_analyzer_parse_options.cpp:48]: I'm not immediately finding one; if you had one in mind can you post a link?

Comment (chrisr-diffblue) [reply src/memory-analyzer/memory_analyzer_parse_options.cpp:48]: Apologies for the chasing of the geese that are wild  (positively livid...), I sort of half-assumed this was referring to the tickets discussing possible refactorings around `languaget` and modes - but I realise now I mis-read this comment.

**Comment by martin-cs** [comment]
@tautschnig @peterschrammel @romainbrenguier you are code-owners for this, please can you review when you have time?  I have removed the controversial commit.

**Review by tautschnig — APPROVED** [review]
Thank you for this cleanup!

**Comment by martin-cs** [comment]
@peterschrammel or @romainbrenguier I need your sign-off for this one.

**Review by peterschrammel — APPROVED** [review]
Lgtm

**Comment by martin-cs** [comment]
Thanks for the fast review.  I will merge when CI passes.

**Pull request merged**
